web/src/stream_create.ts

```diff
@@ -424,7 +424,16 @@ function create_stream(): void {
         },
         error(xhr): void {
             const error_message = z.object({msg: z.string().optional()}).parse(xhr.responseJSON);
-            if (error_message?.msg?.includes("access")) {
+            if (error_message?.msg?.includes("archived")) {
+                stream_name_error.report_already_exists();
+                stream_name_error.select();
+                const message = $t_html({
+                    defaultMessage:
+                        "Error creating channel: An archived channel with this name already exists.",
+                });
+
+                ui_report.error(message, undefined, $(".stream_create_info"));
+            } else if (error_message?.msg?.includes("private")) {
                 // If we can't access the stream, we can safely
```

Comment:
This comment does not seem fully correct now that we do not check whether the string includes "access" now. But am not sure. Maybe can just complete the TODO to use error codes as mentioned in the comment below.

Reply:
Hey @sahil, I don't think we're receiving any error codes in the response. The response always shows `BAD_REQUEST` in the `code` field. Here's the response I get:

```
{"result":"error","msg":"Unable to access private channel (devil).","code":"BAD_REQUEST"}
```

Reply:
Yes, we would need to update the backend code to send different code.